### PR TITLE
docs: clarify where the `async` utility function is coming from

### LINKED ISSUE #34833: Usage of async before-each is wrong or at not described sufficiently
# 📚 Docs or angular.io bug report

### Description

On the Testing Guide, in section 'Component Test Basics/Component DOM testing/CLI generated tests' there is an example that shows an initial spec file for the banner-external component.
This test contains two before-each() declarations, one of them being async. However, there is nothing inside that is being awaited, thus it is very confusing why the async version of before-each() is used or needed. 
The describing text doesn't mention anything either.

### What's the affected URL?**
https://angular.io/guide/testing

### Reproduction Steps**

https://github.com/angular/angular/blob/master/aio/content/guide/testing.md
https://github.com/angular/angular/blob/master/aio/content/examples/testing/src/app/banner/banner-external.component.spec.ts

### Expected vs Actual Behavior**

Actual:
beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ BannerComponent ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(BannerComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

Expected:

  beforeEach(() => {
    TestBed.configureTestingModule({
      declarations: [ BannerComponent ]
    })
    .compileComponents();
    fixture = TestBed.createComponent(BannerComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

## 🔥 Exception or Error
<pre><code>

</code></pre>

### PR DESCRIPTION
**Pull request by sonukapoor**
inside the `beforeEach` for out testing guide.

Fixes #34833

## PR Checklist
Please check if your PR fulfills the following requirements:

- [X] The commit message follows our guidelines: https://github.com/angular/angular/blob/master/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [X] Documentation content changes
- [ ] angular.io application / infrastructure changes
- [ ] Other... Please describe:

## What is the current behavior?

Issue Number: #34833

## Does this PR introduce a breaking change?

- [ ] Yes
- [X] No

### CODE REVIEW

**Comment by sonukapoor** [comment]
@JiaLiPassion / @gkalpak I am not sure about the CI error. I have never seen that before. Can you please help with this?

**Comment by JiaLiPassion** [comment]
@sonukapoor , you don't need to worry about the ci test, it seems a flaky.

File: aio/content/guide/testing.md [hunk 1]
```diff
@@ -618,6 +618,14 @@ It also generates an initial test file for the component, `banner-external.compo
   region="v1"
   header="app/banner/banner-external.component.spec.ts (initial)"></code-example>
 
+<div class="alert is-helpful">
+
+Because `compileComponents` is asynchronous, it uses 
+the [`async`](https://angular.io/api/core/testing/async) utility
```
Comment (JiaLiPassion) [inline aio/content/guide/testing.md]: The link should be (api/core/testing/async)

File: aio/content/guide/testing.md [hunk 2]
```diff
@@ -618,6 +618,14 @@ It also generates an initial test file for the component, `banner-external.compo
   region="v1"
   header="app/banner/banner-external.component.spec.ts (initial)"></code-example>
 
+<div class="alert is-helpful">
+
+Because `compileComponents` is asynchronous, it uses 
+the [`async`](https://angular.io/api/core/testing/async) utility 
+function imported from `@angular/core/testing`.
+
```
Comment (JiaLiPassion) [inline aio/content/guide/testing.md]: Maybe we can also add 
```
please refer to [async](#async) chapter for detail.
```

**Review by JiaLiPassion — CHANGES_REQUESTED** [review]

File: aio/content/guide/testing.md (hunk 1 above)
Comment (sonukapoor) [reply aio/content/guide/testing.md]: @JiaLiPassion, Isn't your link the same as mine?

File: aio/content/guide/testing.md (hunk 2 above)
Comment (sonukapoor) [reply aio/content/guide/testing.md]: Sure, I have added it. Please re-review.

File: aio/content/guide/testing.md (hunk 1 above)
Comment (sonukapoor) [reply aio/content/guide/testing.md]: Good catch! I have fixed this up.

**Review by JiaLiPassion — APPROVED** [review]

**Review by kapunahelewong — APPROVED** [review]
LGTM! Thank you, @sonukapoor!

**Comment by kapunahelewong** [comment]
merge-assistance: Global approval for this docs-only change.

**Pull request closed without merging**